### PR TITLE
Encode `opt_rpitit_info` for associated types

### PR DESCRIPTION
**Pull request by compiler-errors**
Follow-up, only last commit matters

r? @spastorino 

This needs a perf run after the parent pr lands

### CODE REVIEW

File: compiler/rustc_hir_analysis/src/collect/item_bounds.rs [hunk 1]
```diff
@@ -78,6 +78,8 @@ pub(super) fn explicit_item_bounds(
 ) -> &'_ [(ty::Predicate<'_>, Span)] {
     // If the def_id is about an RPITIT, delegate explicit_item_bounds to the opaque_def_id that
     // generated the synthesized associate type.
+    // FIXME(-Zlower-impl-trait-in-trait-to-assoc-ty): This is not correct for bounds with nested
+    // RPITITs.
```
Comment (spastorino) [inline compiler/rustc_hir_analysis/src/collect/item_bounds.rs:82]: What did you mean here exactly?

File: compiler/rustc_metadata/src/rmeta/decoder.rs [hunk 2]
```diff
@@ -1097,7 +1099,7 @@ impl<'a, 'tcx> CrateMetadataRef<'a> {
             container,
             fn_has_self_parameter: has_self,
             // FIXME(-Zlower-impl-trait-in-trait-to-assoc-ty): We need to encode this
```
Comment (spastorino) [inline compiler/rustc_metadata/src/rmeta/decoder.rs:1101]: You probably want to remove the FIXME now?

**Review by spastorino — COMMENTED** [review]
r=me after the other PR lands and perf is good.

**Comment by spastorino** [comment]
@compiler-errors you can already rebase on top of master since your other PR is already in.

File: compiler/rustc_hir_analysis/src/collect/item_bounds.rs (hunk 1 above)
Comment (compiler-errors) [reply compiler/rustc_hir_analysis/src/collect/item_bounds.rs:82]: Maybe I misled myself, will probably just drop this comment and think about it more.

**Comment by compiler-errors** [comment]
Perf neutral

@bors r=spastorino rollup-

**Pull request merged**